### PR TITLE
Feat/fontweight number value

### PR DESCRIPTION
**Pull request by ankit-tailor**
## Summary

This PR adds support for number values for `fontWeight` as requested in #34425. 

## Changelog

[General] [Added] - Added support for number values in fontWeight.

## Test Plan

```js
<Text style={{ fontWeight: 900, color: 'red' }}>
  Hello World
</Text>
```

### CODE REVIEW

File: Libraries/Text/Text.js [hunk 1]
```diff
@@ -176,6 +177,13 @@ const Text: React.AbstractComponent<
     default: accessible,
   });
 
+  let flattenedStyle = flattenStyle(style);
+
+  flattenedStyle =
+    typeof flattenedStyle?.fontWeight === 'number'
+      ? {...flattenedStyle, fontWeight: flattenedStyle?.fontWeight.toString()}
```
Comment (necolas) [inline Libraries/Text/Text.js:184]: I don't think we need to create a new object and could just replace the `fontWeight` value.

Comment (ankit-tailor) [reply Libraries/Text/Text.js]: Yes, made changes.

**Review by necolas — APPROVED** [review]

**Comment by necolas** [comment]
Are there any existing unit tests that we can expand to include this change?

**Comment by ankit-tailor** [comment]
No, I couldn't find any existing tests.

File: Libraries/Text/Text.js [hunk 2]
```diff
@@ -176,6 +177,12 @@ const Text: React.AbstractComponent<
     default: accessible,
   });
 
+  let flattenedStyle = flattenStyle(style);
+
+  if (typeof flattenedStyle?.fontWeight === 'number') {
+    flattenedStyle.fontWeight = flattenedStyle?.fontWeight.toString();
```
Comment (todorone) [inline Libraries/Text/Text.js:183]:
```suggestion
    flattenedStyle.fontWeight = flattenedStyle.fontWeight.toString();
```

We already have null check inside `if` statement

Comment (ankit-tailor) [reply Libraries/Text/Text.js]: Thanks for the suggestions.

**Pull request closed without merging**